src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -254,14 +254,18 @@ def _parse_image_argument(cmd, namespace):
 
     # 4 - attempt to match an URN alias (most likely)
     from azure.cli.command_modules.vm._actions import load_images_from_aliases_doc
-    images = load_images_from_aliases_doc(cmd.cli_ctx)
-    matched = next((x for x in images if x['urnAlias'].lower() == namespace.image.lower()), None)
-    if matched:
-        namespace.os_publisher = matched['publisher']
-        namespace.os_offer = matched['offer']
-        namespace.os_sku = matched['sku']
-        namespace.os_version = matched['version']
-        return 'urn'
+    import requests
+    try:
+        images = load_images_from_aliases_doc(cmd.cli_ctx)
+        matched = next((x for x in images if x['urnAlias'].lower() == namespace.image.lower()), None)
+        if matched:
+            namespace.os_publisher = matched['publisher']
+            namespace.os_offer = matched['offer']
+            namespace.os_sku = matched['sku']
+            namespace.os_version = matched['version']
+            return 'urn'
+    except requests.exceptions.ConnectionError:
+        pass
```

Comment:
we can throw the exception here with the error string you put in the 5th step, since that step won't work anyway 
CC: @qwordy

Reply:
> 
> 
> we can throw the exception here with the error string you put in the 5th step, since that step won't work anyway
> CC: @qwordy

It can be an existing managed disk image resource.

Reply:
@yugangw-msft no, that's what we are trying to avoid here. Even on failure in `#4`, `#5` should still be attempted.